### PR TITLE
Feature/add payment liqpay

### PR DESCRIPTION
**Pull request by sergeyevalex90**
- create pricing component with 3 types of subscription (fornt, back)
- choosing available teachers by subsriptions
- connect and proccesing by liqpay (frontent, backend)
- create and show transactions (back, front)
- pagination (back, front)
- export docs in different formatts

### CODE REVIEW

File: back-end/services/PaymentService.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public_key: publicKey,
+    version: version,
+    action: data.action,
+    amount: data.amount,
+    currency: data.currency,
+    description: data.description,
+    dae: data.dae,
+    order_id: data.order_id,
+    result_url: 'http://localhost:3000/app/finances',
+    server_url: 'https://f951-188-191-234-7.eu.ngrok.io/transactions',
```
Comment (6eluuNiGer) [inline back-end/services/PaymentService.js]: Ngrok only for work localy

File: back-end/services/PaymentService.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  const dataObject = {
+    public_key: publicKey,
+    version: version,
+    action: data.action,
+    amount: data.amount,
+    currency: data.currency,
+    description: data.description,
+    dae: data.dae,
+    order_id: data.order_id,
+    result_url: 'http://localhost:3000/app/finances',
```
Comment (6eluuNiGer) [inline back-end/services/PaymentService.js]: Localhost ?

File: front-end/src/components/Pricing/Pricing.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const [pricing, setPricing] = useState([]);
+  const [teacher, setTeacher] = useState('');
+  const { isAuthenticated, currentUser } = useContext(CurrentUserContext);
+
+  const getPricing = async () => {
+    try {
+      const pricing = await getAllPricing();
+      setPricing(pricing);
+    } catch (error) {
+      console.log(error);
```
Comment (6eluuNiGer) [inline front-end/src/components/Pricing/Pricing.jsx]: remove log

File: front-end/src/components/Pricing/Pricing.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  const handleTeacher = (teacher) => {
+    setTeacher(teacher);
+  };
+
+  useEffect(() => {
+    getPricing();
+  }, []);
+
+  const pricingCards = isAuthenticated && currentUser.role === 'student' && (
+    <React.Fragment>
```
Comment (6eluuNiGer) [inline front-end/src/components/Pricing/Pricing.jsx]: use <> </>

File: front-end/src/components/Pricing/PricingCard/PricingCard.jsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+const PricingCard = ({ pricing, user, teacher }) => {
+  const { name, price, currency, description } = pricing;
+
+  const [payment, setPayment] = useState({
+    data: '',
+    signature: '',
+  });
+
+  const dae = JSON.stringify({
```
Comment (6eluuNiGer) [inline front-end/src/components/Pricing/PricingCard/PricingCard.jsx]: What it is dae?

File: front-end/src/components/Transactions/Transaction/Transaction.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  const userData = JSON.parse(dae);
+
+  // Format timestamp date
+  const cutDate = end_date.substring(0, 10);
+  const date = new Date(cutDate * 1000);
+  const formattedDate = date.getDate() + '.' + (date.getMonth() + 1) + '.' + date.getFullYear();
+
+  return (
+    <tr>
+      <td>{transaction_id}</td>
```
Comment (6eluuNiGer) [inline front-end/src/components/Transactions/Transaction/Transaction.jsx]: Better use  method map

File: front-end/src/components/Transactions/Transactions.jsx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+  const getTransactions = useCallback(async () => {
+    try {
+      const transactions = await getAllTransactions(page);
+      setTransactions(transactions);
+      if (transactions) {
+        setPageCount(transactions.pagination.pageCount);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (6eluuNiGer) [inline front-end/src/components/Transactions/Transactions.jsx]: remove log

File: front-end/src/components/Pricing/PricingCard/PricingCard.jsx (hunk 5 above)
Comment (sergeyevalex90) [reply front-end/src/components/Pricing/PricingCard/PricingCard.jsx]: This is name of the liqpayfield with additional information of the payment

File: back-end/services/PaymentService.js (hunk 1 above)
Comment (sergeyevalex90) [reply back-end/services/PaymentService.js]: Changed to deployed backend url

File: back-end/services/PaymentService.js (hunk 2 above)
Comment (sergeyevalex90) [reply back-end/services/PaymentService.js]: Change to deployed frontend url

File: front-end/src/components/Pricing/Pricing.jsx (hunk 3 above)
Comment (sergeyevalex90) [reply front-end/src/components/Pricing/Pricing.jsx]: Done

File: front-end/src/components/Pricing/Pricing.jsx (hunk 4 above)
Comment (sergeyevalex90) [reply front-end/src/components/Pricing/Pricing.jsx]: Done

File: front-end/src/components/Pricing/PricingCard/PricingCard.jsx (hunk 5 above)
Comment (sergeyevalex90) [reply front-end/src/components/Pricing/PricingCard/PricingCard.jsx]: Changed dae to userInfo

File: front-end/src/components/Transactions/Transaction/Transaction.jsx (hunk 6 above)
Comment (sergeyevalex90) [reply front-end/src/components/Transactions/Transaction/Transaction.jsx]: Done

File: front-end/src/components/Transactions/Transactions.jsx (hunk 7 above)
Comment (sergeyevalex90) [reply front-end/src/components/Transactions/Transactions.jsx]: Done

**Review by 6eluuNiGer — APPROVED** [review]

**Pull request merged**